src/lib/media-import/status.js

```diff
@@ -0,0 +1,230 @@
+/**
+ * @flow
+ * @format
+ */
+
+/**
+ * External dependencies
+ */
+import chalk from 'chalk';
+import gql from 'graphql-tag';
+import debugLib from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import API from 'lib/api';
+import { currentUserCanImportForApp } from 'lib/media-import/media-file-import';
+import { MediaImportProgressTracker } from 'lib/media-import/progress';
+import { capitalize, formatEnvironment, getGlyphForStatus } from 'lib/cli/format';
+import { ProgressTracker } from '../cli/progress';
+
+const debug = debugLib( 'vip:lib/media-import/status' );
+
+const IMPORT_MEDIA_PROGRESS_POLL_INTERVAL = 5000;
+
+/*
+ * TODO: Add support to
+ *  failureDetails {
+ *		previousStatus
+ *		globalErrors
+ *	}
+ * (when Parker is ready)
+ */
+const IMPORT_MEDIA_PROGRESS_QUERY = gql`
+	query App($appId: Int = 1, $envId: Int = 1) {
+		app(id: $appId) {
+			environments(id: $envId ) {
+			id
+			name
+			type
+			repo
+			mediaImportStatus {
+				importId
+				siteId
+				status
+				filesTotal
+				filesProcessed
+			}
+		}
+	}
+}
+`;
+
+export type MediaImportCheckStatusInput = {
+	app: Object,
+	env: Object,
+	MediaImportProgressTracker: MediaImportProgressTracker,
+};
+
+async function getStatus( api, appId, envId ) {
+	const response = await api.query( {
+		query: IMPORT_MEDIA_PROGRESS_QUERY,
+		variables: { appId, envId },
+		fetchPolicy: 'network-only',
+	} );
+
+	const {
+		data: {
+			app: { environments },
+		},
+	} = response;
+	if ( ! environments?.length ) {
+		throw new Error( 'Unable to determine import status from environment' );
+	}
+	const [ environment ] = environments;
+	const { mediaImportStatus } = environment;
+
+	return mediaImportStatus;
+}
+
+function buildErrorMessage( importFailed ) {
```

Comment:
This is the log on the FAILED status
```
=============================================================
Checking the Media import status for your environment...

=============================================================
Site ID: 1000
Current status: FAILED
Processed Files: 100/300 - 33%
=============================================================
=============================================================
Status: Checking...
Site: test1-go-vip (PRODUCTION)
=============================================================


Error: FAILED
An error occurred: 100/300 files imported.

If this error persists, please contact support.
```

1. I'm not sure if the direct response from Parker should be displayed. Maybe just leave a generic `Import #1 failed` for now?
2. The processed files percentage was already displayed , so maybe it is redundant
3. There are two status fields and one of them is `Checking...` . Also should we get rid of all fields that are already getting displayed?

Reply:
> 1. I'm not sure if the direct response from Parker should be displayed. Maybe just leave a generic Import #1 failed for now?

Probably not, although I think that errors returned from Parker should not contain any private or sensitive data, but should instead be of help to users to understand what the underlying problem is.

> 2. The processed files percentage was already displayed , so maybe it is redundant

Where it is already displayed? I think it appears only once, or? 🤔 

> 3. There are two status fields and one of them is Checking... . Also should we get rid of all fields that are already getting displayed?

Yeah, that may be confusing. `Current Status` was intended to display the current import state i.e. `INITIALIZING`, `COMPLETING`,  etc. I guess we could just rename it to something else.

Reply:
For the last point, I have for now reworded to `Current step` although I don't like it because it's more of state... a79abdf
Perhaps I should just rename the other `Status`. 🤔

Reply:
Repeated output fixed.